elastic_transport/_async_transport.py

```diff
@@ -174,7 +174,7 @@ def __init__(
         # sniffing. Uses '_sniffing_task' instead.
         self._sniffing_lock = None  # type: ignore[assignment]
 
-    async def perform_request(  # type: ignore[override]
```

Comment:
I assume we do not care that the public interface of this method changes? This is called by the client library package (which we are also updating in parallel)? Any other possible callers that may break with this change?

Reply:
In general we do care about the public interface, which is used by projects like https://github.com/elastic/enterprise-search-python, https://github.com/elastic/elasticsearch-serverless-python and https://github.com/ywangd/peek/. But:

 * We don't care about removing `endpoint_id` and `path_parts` from the signature because this change was never released.
 * We don't care about adding `otel_span` because it is a new keyword argument with a default value, and thus not breaking.